### PR TITLE
Cleanup platformBomXXX and templates

### PR DESCRIPTION
**Pull request by maxandersen**
Why:

 * maven and gradle uses various different spelling and casing of
   quarkusPlatformBomXXX

This change addreses the need by:

 * search and replace to consistently use `quarkusPlatformXXX` (i.e not bom)
   references as the data is intended to be used to resolve the "platform"
   rather than a specific resource.
 * Future improvements can be done in pom.xml and gradle build files to
   allow overrides but changing that this close to release I evaluated as
   being a bad idea :)

### CODE REVIEW

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by maxandersen** [comment]
lets wait for the CI build to confirm as it could break stuff if I missed a space ;)

**Comment by gastaldi** [comment]
Too bad my review is worthless for CI, let's wait until someone with privileges to approve @geoand ?

**Review by geoand — APPROVED** [review]
LGTM

**Pull request merged**
